lib/Optimizer/GraphOptimizer.cpp

```diff
@@ -1076,14 +1076,22 @@ static void optimizeSliceOfSplat(Function *F) {
 }
 
 /// Eliminate ReshapeNode when the input is already the correct shape.
+/// Also eliminate ReshapeNode when the input is SplatNode.
 static void optimizeReshape(Function *F) {
   for (auto &node : F->getNodes()) {
     auto *reshapeNode = dyn_cast<ReshapeNode>(&node);
     if (!reshapeNode)
       continue;
-    auto inputNode = reshapeNode->getInput();
+    auto &inputNode = reshapeNode->getNthInput(0);
     if (inputNode.dims() == reshapeNode->dims()) {
       reshapeNode->getResult().replaceAllUsesOfWith(inputNode);
+      continue;
+    }
+    auto *splatNode = dyn_cast<SplatNode>(inputNode);
```

Comment:
Add a comment before this line to describe the transformation, e.g.:
```c++
// Reshape(Splat(args)) -> Splat(args')
```

Reply:
Done.